src/index.js

```diff
@@ -1,21 +1,27 @@
 const { useEffect } = require('react');
+const invariant = require('invariant');
 const { onKeyPress, convertToAsciiEquivalent, getAsciiCode } = require('./keys.js');
 
-const useKey = (callback, { detectKeys = [] } = {}, { dependencies = [] } = {}) => {
-  let allowedKeys = detectKeys;
-  if (!window || !window.document || !callback) {
-    throw new Error();
-  }
+const VALID_KEYEVENTS = ['keydown', 'keyup', 'keypress'];
 
-  if (!Array.isArray(dependencies)) {
-    throw new Error(typeof dependencies);
-  }
+const useKey = (callback, { detectKeys = [], keyevent = 'keydown' } = {}, { dependencies = [] } = {}) => {
+  const isKeyeventValid = VALID_KEYEVENTS.indexOf(keyevent) > -1;
+
+  invariant(isKeyeventValid, 'keyevent is not valid: ' + keyevent);
+  invariant(callback != null, 'callback needs to be defined');
+  invariant(window != null, 'window needs to be defined');
+  invariant(window.document != null, 'window.document needs to be defined');
+  invariant(Array.isArray(dependencies), 'dependencies need to be an array');
+  invariant(Array.isArray(dependencies), 'dependencies need to be an array');
+
+  let allowedKeys = detectKeys;
 
   if (!Array.isArray(detectKeys)) {
     allowedKeys = [];
     // eslint-disable-next-line no-console
     console.warn('Keys should be array!');
   }
+
   allowedKeys = convertToAsciiEquivalent(allowedKeys);
 
   const handleKeydown = event => {
```

Comment:
Now that this function is going to be common for all 3 events, let's name it better.
`handleEvent` or some other meaningful name.
WYT?

Reply:
Ah yes! That makes sense, definitely a good idea. Can you change this when merging?